src/main/java/org/manifold/compiler/front/StaticAttributeAccessVertex.java

```diff
@@ -0,0 +1,111 @@
+package org.manifold.compiler.front;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.manifold.compiler.TypeValue;
+import org.manifold.compiler.Value;
+
+import com.google.common.base.Preconditions;
+
+public class StaticAttributeAccessVertex extends ExpressionVertex {
+
+  private static Logger log = LogManager.getLogger(
+      "StaticAttributeAccessVertex");
+
+  private final ExpressionEdge exprEdge;
+  public ExpressionEdge getExpressionEdge() {
+    return this.exprEdge;
+  }
+
+  private final String attributeID;
+  public String getAttributeID() {
+    return this.attributeID;
+  }
+
+  public StaticAttributeAccessVertex(ExpressionGraph exprGraph,
+      ExpressionEdge exprEdge, String attrID) {
+    super(exprGraph);
+    this.exprEdge = exprEdge;
+    this.exprEdge.setTarget(this);
+    this.exprEdge.setName("expr");
+    this.attributeID = attrID;
+  }
+
+  private TypeValue type = null;
+
+  @Override
+  public TypeValue getType() {
+    return type;
+  }
+
+  private Value value = null;
+
+  @Override
+  public Value getValue() {
+    return this.value;
+  }
+
+  @Override
+  public void elaborate() throws Exception {
+    log.debug("elaborating static attribute access");
```

Comment:
nit: I believe we can add ```if (value != null && type != null) return;``` to avoid duplicate elaborations, similar to the other vertices.